keanu-project/src/main/java/io/improbable/keanu/util/io/DotSaver.java

```diff
@@ -59,45 +61,64 @@ public DotSaver(BayesianNetwork network) {
     @Override
     public void save(OutputStream output, boolean saveValues, Map<String, String> metadata) throws IOException {
         Preconditions.checkArgument(bayesianNetwork.getAllVertices().size() > 0, "Network must contain at least one vertex.");
-        Vertex anyVertex = bayesianNetwork.getAllVertices().get(0);
-        save(output, anyVertex, INFINITE_NETWORK_DEGREE, saveValues, metadata);
+        List<Vertex> anyVertices = getAnyVertexFromEachConnectedGraph();
+        save(output, anyVertices, INFINITE_NETWORK_DEGREE, saveValues, metadata);
+    }
+
+    private List<Vertex> getAnyVertexFromEachConnectedGraph() {
+        List<Vertex> anyVertices = new ArrayList<>();
+        List<Vertex> processedVertices = new ArrayList<>();
+        List<Vertex> allVertices = bayesianNetwork.getAllVertices();
+
+        for (Vertex vertex : allVertices) {
+            if (!processedVertices.contains(vertex)) {
+                anyVertices.add(vertex);
+                processedVertices.addAll(vertex.getConnectedGraph());
+            }
+        }
+
+        return anyVertices;
     }
 
     /**
-     * Outputs a subgraph around the specified vertex to a DOT file which can be used by various graph visualizers to generate a visual representation of the graph.
+     * Outputs a subgraph around the specified vertices to a DOT file which can be used by various graph visualizers to generate a visual representation of the graph.
      * Read more about DOT format here: https://en.wikipedia.org/wiki/DOT_(graph_description_language)
      *
      * @param output     output stream to use for writing
-     * @param vertex     vertex around which the subgraph will be centered
+     * @param vertices   vertices around which the subGraphs will be centered
      * @param degree     degree of connections to be visualized; for instance, if the degree is 1,
-     *                   only connections between the vertex and its parents and children will be written out to the DOT file.
+     *                   only connections between the vertices and its parents and children will be written out to the DOT file.
      * @param saveValues specify whether you want to output values of non-constant scalar vertices
      * @throws IOException Any errors that occur during saving to the output stream
      */
-    public void save(OutputStream output, Vertex vertex, int degree, boolean saveValues) throws IOException {
-        save(output, vertex, degree, saveValues, null);
+    public void save(OutputStream output, List<Vertex> vertices, int degree, boolean saveValues) throws IOException {
```

Comment:
Add a helper `save` method that takes a single Vertex and creates a `Collections.singletonList()` and calls into this method. It's annoying in your tests having to make lists all the time for the use case of a single vertex